packages/redux/src/analytics/middlewares/__tests__/bag.test.ts

```diff
@@ -393,80 +396,119 @@ describe('analyticsBagMiddleware', () => {
       });
     });
 
-    it('Should track a remove and an add event if the product sizes changed', async () => {
-      // eslint-disable-next-line @typescript-eslint/no-var-requires
-      const { analyticsBagMiddleware } = require('../bag');
+    it('should trigger only PRODUCT_UPDATED event when it changed size event, with sizeId property assigned on event payload.', async () => {
+      const size = {
+        ...(bagMockData.mockSizes[1] as SizeAdapted),
+        id: 999,
+        name: 'XXL',
+      };
 
-      const simplifiedStore = mockSimplifiedStore(bagMockData.mockState, [
-        analyticsBagMiddleware(analytics),
-      ]);
+      const bagItem = bagMockData.mockBagItem().getItem({ size });
 
-      await simplifiedStore.dispatch({
+      await store.dispatch({
         type: bagActionTypes.UPDATE_BAG_ITEM_SUCCESS,
         payload: {
           result: { id: bagMockData.mockBagId },
           entities: {
+            size: {
```

Comment:
Does this entity exist? Looking at the bag item schema this does not seem to be created